api/src/main/java/org/openmrs/api/impl/PersonServiceImpl.java

```diff
@@ -487,28 +488,20 @@ public Relationship saveRelationship(Relationship relationship) throws APIExcept
 	 *      java.lang.String)
 	 */
 	@Override
-	public Relationship voidRelationship(Relationship relationship, String voidReason) throws APIException {
-		if (relationship.getVoided()) {
-			return relationship;
-		}
-		
-		relationship.setVoided(true);
-		if (relationship.getVoidedBy() == null) {
-			relationship.setVoidedBy(Context.getAuthenticatedUser());
-		}
-		if (voidReason != null) {
-			relationship.setVoidReason(voidReason);
+	public Relationship voidRelationship(@NonNull final Relationship relationship, String voidReason) throws APIException {
+		if (relationship ==null) {
+			return null;
```

Comment:
it's not a good practical way to directly return null, this might throw nullpointer Exceptions .[feel free to checkout here](https://stackoverflow.com/questions/1274792/is-returning-null-bad-design)

Reply:
@sherrif10  thanks let check agaiin